File: libraries/hlp/insub.f
      INTEGER FUNCTION hlp_INSUB (STRING, PROMPT, L)
*+
*  - - - - - -
*   I N S U B
*  - - - - - -
*
*  A minimalist example of the user-supplied INSUB routine, which is
*  called by the hlp_HELP routine to obtain an interactive response.
*
*
*  Returned:
*     STRING      c*(*)    response string
*
*  Given:
*     PROMPT      c*(*)    prompt string
*
*  Returned:
*     L           i        length of string excluding trailing blanks
*     hlp_INSUB   i        status: always 1=OK
*
*  Called:  hlp_LENGTH
*
*  I/O units:
*     LUCMD   input    command
*     LUMES   output   prompt
*
*  Note:  The calling sequence for this routine is the same as that for
*         the VAX/VMS routine LIB$GET_INPUT.
*
*  P.T.Wallace   Starlink   28 July 1992
*-

      IMPLICIT NONE

      CHARACTER*(*) STRING,PROMPT
      INTEGER L

*  I/O unit numbers
      INTEGER LUCMD,LUMES
      PARAMETER (LUCMD=5,LUMES=6)     !!! Machine dependent !!!

      INTEGER hlp_LENGTH


      WRITE (LUMES,'(A,$)') PROMPT     !!! Sun SPARC etc
      READ (LUCMD,'(A)') STRING
      L=hlp_LENGTH(STRING)
      hlp_INSUB=1

      END

File: libraries/hlp/insub.c
#include "help.h"

int hlpInsub ( char *string, char *prompt, int *l )
/*
**  - - - - - - - - -
**   h l p I n s u b
**  - - - - - - - - -
**
**  A minimalist example of the user-supplied insub routine, which is
**  called by the hlpHelp routine to obtain an interactive response.
**
**  Input is from stdin and output to stdout.
**
**  Returned:
**     *string     char     response string (up to 80 characters)
**
**  Given:
**     *prompt     char     prompt string
**
**  Returned (argument):
**     *l          int      length of string excluding trailing blanks
**
**  Returned (function value):
**                 int      always +1
**
**  Called:  hlpLength
**
**  Last revision:   30 July 2009
**
**  Copyright P.T.Wallace.  All rights reserved.
*/
{
   int n, c;
   char *cs;


/* Output the prompt string */
   fputs ( prompt, stdout );

/* Read the input string, stopping at and eliminating any newline */
   cs = string;
   for ( n = 80 ; --n > 0 && ( c = getchar ( ) ) != EOF ; ) {
      if ( c != '\n' ) {
         *cs++ = (char) c;
      } else {
         break;
      };
    };

/* Terminate the string */
   *cs = '\0';

/* Determine the length excluding white space */
   *l = hlpLength ( string );

   return 1;
}
